src/wp-includes/l10n.php

```diff
@@ -1190,104 +1217,59 @@ function load_script_translations( $file, $handle, $domain ) {
  * @since 4.6.0
  * @access private
  *
- * @see get_translations_for_domain()
- * @global MO[] $l10n_unloaded An array of all text domains that have been unloaded again.
+ * @global MO[]                   $l10n                   An array of all currently loaded text domains.
+ * @global MO[]                   $l10n_unloaded          An array of all text domains that have been unloaded again.
+ * @global WP_Textdomain_Registry $wp_textdomain_registry WordPress Textdomain Registry.
  *
  * @param string $domain Text domain. Unique identifier for retrieving translated strings.
  * @return bool True when the textdomain is successfully loaded, false otherwise.
  */
 function _load_textdomain_just_in_time( $domain ) {
-	global $l10n_unloaded;
+	global $l10n, $l10n_unloaded, $wp_textdomain_registry;
 
 	$l10n_unloaded = (array) $l10n_unloaded;
 
-	// Short-circuit if domain is 'default' which is reserved for core.
-	if ( 'default' === $domain || isset( $l10n_unloaded[ $domain ] ) ) {
+	if ( isset( $l10n_unloaded[ $domain ] ) ) {
 		return false;
 	}
 
-	$translation_path = _get_path_to_translation( $domain );
-	if ( false === $translation_path ) {
-		return false;
+	/** @var WP_Textdomain_Registry $wp_textdomain_registry */
+	if ( null === $wp_textdomain_registry->get( $domain ) ) {
+		$wp_textdomain_registry->get_translation_from_lang_dir( $domain );
 	}
 
-	return load_textdomain( $domain, $translation_path );
```

Comment:
`load_textdomain()` is no longer used because it calls `$wp_textdomain_registry->set()`?

I think not calling `load_textdomain()` will cause some issues because the function has a few actions/filters which will no longer be available.

Reply:
Agreed. Could probably just do this:

```php

$path = $wp_textdomain_registry->get( $domain );

$locale = is_admin() ? get_user_locale() : get_locale();
$mofile = "{$path}/{$domain}-{$locale}.mo";

if ( 'default' === $domain ) {
	$mofile = "{$path}/{$locale}.mo";
}

return load_textdomain( $domain, $path );
```